src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java

```diff
@@ -0,0 +1,209 @@
+
+    // must be done as the last step to catch all imports
+    apiFile.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+
+    return apiFile.build();
+  }
+
+  private StaticLangApiMessageView generateSchemaClass(
+      DiscoGapicInterfaceContext context,
+      String schemaName,
+      Schema schema,
+      SymbolTable schemaSymbolTable) {
+    addApiImports(context);
+
+    StaticLangApiMessageView.Builder schemaView = StaticLangApiMessageView.newBuilder();
+
+    schemaView.typeName(schemaName);
+    schemaView.type(schema.type());
+    // TODO(andrealin): apply Java naming format.
+    // TODO(andrealin): use symbol table to make sure Schema names aren't Java keywords.
+    schemaView.defaultValue(schema.defaultValue());
+
+    // Map each property name to the Java typeName of the property.
+    List<SimplePropertyView> properties = new LinkedList<>();
+    for (Map.Entry<String, Schema> propertyEntry : schema.properties().entrySet()) {
+      String propertyString = schemaSymbolTable.getNewSymbol(propertyEntry.getKey());
+      Schema property = propertyEntry.getValue();
+      SimplePropertyView.Builder simpleProperty =
+          SimplePropertyView.newBuilder().name(propertyString);
+      simpleProperty.typeName(typeToJavaType(property));
+
+      // TODO(andrealin) use a surface namer for the getter/setter.
+      FieldCopyView.Builder getterAndSetter = FieldCopyView.newBuilder();
+      Name propertyName = Name.anyCamel(propertyString);
+      getterAndSetter.fieldGetFunction(
+          nameFormatter.publicMethodName(Name.from("get").join(propertyName)));
+      getterAndSetter.fieldSetFunction(
+          nameFormatter.publicMethodName(Name.from("set").join(propertyName)));
+      simpleProperty.fieldCopyView(getterAndSetter.build());
+      properties.add(simpleProperty.build());
+    }
+    schemaView.properties(properties);
+
+    return schemaView.build();
+  }
+
+  private void addApiImports(DiscoGapicInterfaceContext context) {
+    ModelTypeTable typeTable = context.getModelTypeTable();
+    typeTable.saveNicknameFor("com.google.api.core.BetaApi");
+    typeTable.saveNicknameFor("java.io.Serializable");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("javax.annotation.Generated");
+  }
+
+  // Return the corresponding Java identifier for a given Discovery doc typeName and format.
+  // https://developers.google.com/discovery/v1/type-format.
+  private String typeToJavaType(Schema schema) {
+    if (!schema.reference().isEmpty()) {
+      return schema.reference();
+    }
+
+    switch (schema.type()) {
+      case ARRAY:
+        return String.format("List<%s>", typeToJavaType(schema.items()));
+      case INTEGER:
+        switch (schema.format()) {
+          case INT32:
+            return "Integer";
+          case UINT32:
+            return "Long";
+          default:
+            System.err.println("Discovery doc had an INTEGER typeName that was not Integer/Long.");
```

Comment:
Change your println's to exceptions for now (`IllegalStateException` is fine) - we want to know early when we aren't handling cases that we are supposed to.